protocols/dcutr/src/behaviour.rs

```diff
@@ -86,11 +87,26 @@ impl Behaviour {
 impl NetworkBehaviour for Behaviour {
     type ConnectionHandler = handler::Prototype;
     type OutEvent = Event;
+    type DialPayload = handler::Prototype;
```

Comment:
This is a hack for now, the idea would be not pass the handler here but just a boring data structure and then construct the handler inside the callback.